ui/src/timeMachine/components/TimeMachineFluxEditor.tsx

```diff
@@ -56,31 +56,58 @@ const TimeMachineFluxEditor: FC<Props> = ({
     onSetActiveQueryText(editorInstance.getValue())
   }
 
+  const getInsertLineNumber = (
+    currentLineNumber: number,
+    scriptLines: string[]
+  ): number => {
+    const currentLine =
+      scriptLines[currentLineNumber] || scriptLines[scriptLines.length - 1]
+
+    // Insert on the current line if its an empty line
+    if (!currentLine.trim()) {
+      return currentLineNumber
+    }
+
+    return currentLineNumber + 1
+  }
+
   const handleInsertFluxFunction = (func: FluxToolbarFunction): void => {
     const p = editorInstance.getPosition()
+    const scriptLines = activeQueryText.split('\n')
+
+    const insertLineNumber = getInsertLineNumber(p.lineNumber, scriptLines)
+
     // sets the range based on the current position
     let range = new window.monaco.Range(
-      p.lineNumber,
-      p.column,
-      p.lineNumber,
-      p.column
+      insertLineNumber, // the row beneath the cursor
+      1, // beginning column of the row
+      insertLineNumber,
+      1
     )
     // edge case for when user toggles to the script editor
     // this defaults the cursor to the initial position (top-left, 1:1 position)
+    const [currentRange] = editorInstance.getVisibleRanges()
+    // Determines whether the new insert line is beyond the current range
+    let insertOnLastLine = insertLineNumber > currentRange.endLineNumber
```

Comment:
strongly recommend this variable be changed to something like `shouldInsertOnLastLine`.

right now, there's a variable called `insertLineNumber` that represents the line to insert a function on, and there's this, called `insertOnLastLine` which is a boolean. The similar names with different values and different semantic meanings is really confusing. Either of these names could also be understood to be verbs, and so someone glancing at this might think they're methods, which is another layer of confusion. These are pretty well named variables other than they're a little bit ambiguous in their intended usage. I don't know a clever way to be precise, so I usually just do it with a long variable name. `lineNumberToInsertAt` vs `insertLineNumber`.